TypeCobol/Compiler/CodeElements/CodeElement.cs

```diff
@@ -157,16 +159,20 @@ public override string ToString()
 
         public override bool Equals(object obj)
         {
-            var codeElement = obj as CodeElement;
-            if (codeElement == null)
-                return false;
-
-            return this.Line == codeElement.Line &&
-                   this.Type == codeElement.Type &&
-                   this.Column == codeElement.Column &&
-                   this.StartIndex == codeElement.StartIndex &&
-                   this.StopIndex == codeElement.StopIndex &&
-                   this.Text == codeElement.Text;
+            return Equals(obj as CodeElement);
+        }
+
+        public bool Equals(CodeElement codeElement)
+        {
+            if (Object.ReferenceEquals(this, codeElement)) return true;
+            if (codeElement == null) return false;
+
+            return Type == codeElement.Type && ConsumedTokens.Equals(codeElement.ConsumedTokens);
```

Comment:
I think Equals on list will only compare reference of list not their content.
As CodeElement cannot overlaps, I think we only need to compare the first token of each ConsumedTokens list.

Reply:
It's right; it's modified by using ConsumedTokens[0] if it exists.
https://github.com/TypeCobolTeam/TypeCobol/pull/1555/commits/ca8920a3aabefc6b8a0d435082b58f0dbb1d4949